os/emscripten/pre.js

```diff
@@ -71,6 +71,33 @@ Module.preRun.push(function() {
          *  add_server("localhost", 3979); */
     }
 
+    var leftButtonDown = false;
+    document.addEventListener("mousedown", e => {
+	if (e.button == 0) {
+	    leftButtonDown = true;
+	}
+    });
+    document.addEventListener("mouseup", e => {
+	if (e.button == 0) {
+	    leftButtonDown = false;
+	}
+    });
+    window.openttd_open_url = function(url, url_len) {
+	const url_string = UTF8ToString(url, url_len);
+	function openWindow() {
+	    document.removeEventListener("mouseup", openWindow);
+	    window.open(url_string, '_blank');
+	}
+	/* Trying to open the URL while the mouse is down results in the button getting stuck, so wait for the
+	 * mouse to be released before opening it. However, when OpenTTD is lagging, the mouse can get released
+	 * before the button click even registers, so check for that, and open the URL immediately if that's the
+	 * case. */
+        if (leftButtonDown) {
+            document.addEventListener("mouseup", openWindow);
+	} else
+            openWindow();
```

Comment:
You forgot the `else` case :)

Also, please checkout out your spaces here, it seems to vary a lot :P Seems this file has 4 spaces for indentation, so it would be good to apply this to all new code here. But mostly, the `if` has a different indentation than the `else` :D

Reply:
The weirdness is probably VSCode choking on the mixed indentation; I'll hand-edit it to be consistent.